zegami_sdk/collection.py

```diff
@@ -291,6 +305,104 @@ def imageset_index_to_row_index(self, imageset_index, source=0) -> int:
             raise IndexError(
                 "Invalid imageset index {} for this source"
                 .format(imageset_index))
+    
+    def add_snapshot(self, name, desc, snapshot):
+        url = '{}/{}/project/{}/snapshots/{}/snapshots'.format(
+            self.client.HOME, self.client.API_0, self.workspace_id, self.id)
+        payload = {
+            'name': name,
+            'description': desc,
+            'snapshot': snapshot,
+            'version': 3,
+        }
+        r = self.client._auth_post(url, json.dumps(payload), return_response=True)
+        return r
+
+    def add_feature_pipeline(self, name, steps, source=0, generate_snapshot=False):
+        # get the source
+        source = self._parse_source(source)
+        node_group = [
+            'source_{}'.format(source.name),
+            'collection_{}'.format(self.id),
+        ]
+
+        mRMR_params = steps[0]['params']
+
+        # find the feature extraction node
+        source_feature_extraction_node = self.get_feature_extraction_imageset_id(source)
+
+        join_dataset_id = source._imageset_dataset_join_id
+        imageset_parents = [source_feature_extraction_node]
+        dataset_parents = [self._dataset_id, join_dataset_id]
+
+        mRMR_node = add_node(
+            self.client,
+            self.workspace,
+            action=steps[0]['action'],
+            params=mRMR_params,
+            type="imageset",
+            dataset_parents=dataset_parents,
+            imageset_parents=imageset_parents,
+            processing_category='image_clustering',
+            node_group=node_group,
+            name="mRMR imageset for {} of {}".format(source.name, self.name),
```

Comment:
The node names should incorporate the name the user passed into this function.

Reply:
Although what's the point of this? I don't think they can check the name of this node in the client 0.0

Reply:
Even for me debugging through Postman it would be helpful for them to have unique/descriptive names 😄 .
Possibly not long before we expose these node names in the Collection Inspector too.